test/then.cc

```diff
@@ -42,6 +44,20 @@ TEST(ThenTest, Succeed) {
   EXPECT_EQ("then", *c());
 }
 
+TEST(ThenTest, SucceedVoid) {
+  bool ran = false;
+  auto e = [&]() {
+    return Just()
+        | Then([&]() {
+             ran = true;
+           });
+  };
+
+  // A Then<void>() doesn't return anything.
```

Comment:
So as not to confuse folks that `Then` can take a `void` type parameter, can I suggest something like:
```suggestion
  // Passing a callable (e.g., a lambda) that doesn't return anything to 'Then()' also doesn't return anything.
```

Reply:
Done, with a phrasing tweak:

```
  // Then() doesn't return anything when its callable (in this case, its
  // lambda) doesn't return anything.
```